xfer.py

```diff
@@ -1,743 +1,754 @@
-                    key, val = entry.split("=")
-                    if key == 'timestamp':
-                        continue
-                    sync_request[key] = int(val)
-            elif info[0] == 'TRANSFER_REQUEST' or info[0] == 'VERIFY_REQUEST': # Format: TRANSFER_REQUEST fname size
-                if sync_request_start is None:
-                    logging.error("Transfer request found at line %d before sync started; inconsistent log", idx)
-                    sys.exit(4)
-                if info[1][0] == '{':
-                    local_info = json.loads(" ".join(info[1:]))
-                    size = int(local_info['size'])
-                    fname = local_info['name']
-                else:
-                    size = int(info[2])
-                    fname = info[1]
-                # File was previously verified.
-                if fname in sync_request['verified_files'] and sync_request['verified_files'][fname] == size:
+    for idx, line in enumerate(transfer_manifest.open(mode="r")):
+        info = line.strip().split()
+        # Format: SYNC_REQUEST {} files_at_source={} files_to_transfer={} bytes_to_transfer={} files_to_verify={} bytes_to_verify={} timestamp={}
+        if info[0] == "SYNC_REQUEST":
+            sync_count += 1
+            # if sync_request_start is not None:
+            #    logging.error("Sync request started at line %d but never finished; inconsistent log",
+            #        sync_request_start)
+            #    sys.exit(4)
+            sync_request_start = idx
+            for entry in info[2:]:
+                key, val = entry.split("=")
+                if key == "timestamp":
                     continue
-                sync_request['files'][fname] = size
-                if info[0] == 'TRANSFER_REQUEST':
-                    if info[1][0] == '{':
-                        local_info = json.loads(" ".join(info[1:]))
-                        sync_request['xfer_files'].add(local_info['name'])
-                    else:
-                        sync_request['xfer_files'].add(info[1])
-            elif info[0] == 'TRANSFER_VERIFIED': # Format: TRANSFER_VERIFIED relative_fname hexdigest size timestamp:
-                if sync_request_start is None:
-                    logging.error("Transfer verification found at line %d before sync started; inconsistent log", idx)
-                    sys.exit(4)
-                if info[1][0] == '{':
+                sync_request[key] = int(val)
+        # Format: TRANSFER_REQUEST fname size
+        elif info[0] == "TRANSFER_REQUEST" or info[0] == "VERIFY_REQUEST":
+            if sync_request_start is None:
+                logging.error(
+                    "Transfer request found at line %d before sync started; inconsistent log",
+                    idx,
+                )
+                sys.exit(4)
+
+            local_info = json.loads(" ".join(info[1:]))
+            size = int(local_info["size"])
+            fname = local_info["name"]
```

Comment:
No backwards compatibility with reading non-JSON entries? This is ok, we can regenerate existing `transfer_manifest.txt`s (though that can take a while), just want to make sure I'm not missing something.

Reply:
Yep, no backwards compat. Perhaps we can write a script to do the conversion just one time? I'd rather not have compat code lying around forever.

Reply:
One-off script sounds good to me.